add api endpoint /users

larsbo opened the pull request:
## Description

This is my first larger pull request - I hope I didn't mess it up totally 
There is one part I'm unsure about: It looks like there is no well matching bundle for the new user validation services. I chose the account bundle but perhaps a new one would make more sense?

| Questions               | Answers |
|-------------------------|-------------------------------------------------------|
| Why?                    | This PR adds the REST API endpoint `/users` for manipulating backend users. |
| BC breaks?              | no |
| Tests exists & pass?    | yes |
| Related tickets?        | - |
| How to test?            | Running PHPUnit tests. |
| Requirements met?       | I hope so :) |

jenskueper commented:
Hi @larsbo,

thank you for your work and the extensive PR! I tested it a bit and one thing that appears to be not working is the automatic password generation. I created a new user and tried the login with the generated password but I had no success. Could you check this?

larsbo commented:
@jenskueper thanks for reviewing/testing :)

I fixed the password bug and also added a user name validator because there was no duplicate check so far.

mitelg commented:
thanks for your contribution @larsbo :+1: :tada:

pweyck commented:
Thanks for this addition! 

We've changed the event name and added tests and a change log entry.
